keps/sig-apps/4368-support-managed-by-label-for-batch-jobs/README.md

```diff
@@ -0,0 +1,985 @@
+  CRI or CNI may require updating that component before the kubelet.
+-->
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+<!--
+This section must be completed when targeting alpha to a release.
+-->
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+<!--
+Pick one of these and delete the rest.
+
+Documentation is available on [feature gate lifecycle] and expectations, as
+well as the [existing list] of feature gates.
+
+[feature gate lifecycle]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[existing list]: https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/
+-->
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: `JobManagedByLabel`
+  - Components depending on the feature gate: `kube-apiserver`, `kube-controller-manager`
+- [ ] Other
+  - Describe the mechanism:
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node?
+
+###### Does enabling the feature change any default behavior?
+
+No.
+
+<!--
+Any change of default behavior may be surprising to users or break existing
+automations, so be extremely careful here.
+-->
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes.
+
+<!--
+Describe the consequences on existing workloads (e.g., if this is a runtime
+feature, can it break the existing applications?).
+
+Feature gates are typically disabled by setting the flag to `false` and
+restarting the component. No other changes should be necessary to disable the
+feature.
+
+NOTE: Also set `disable-supported` to `true` or `false` in `kep.yaml`.
+-->
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The feature behaves as if it was enabled for the first time.
```

Comment:
Can we mention that a conflict can happen between the controllers when a label was set here as well?

Reply:
Could you explain the scenario, what type of conflict?

Reply:
to just link or highlight the `Two controllers running at the same time on old version` section again

Reply:
We may get to controllers running when the feature is disabled, but IIUC this question is about the situation when the feature enabled (re-enabled). To enable (or re-enable) the feature it needs to be already supported, so the version isn't "old", as in the risk section.

Reply:
yeah, I meant only the newly added part

```
Note that an analogous situation may happen when the version of Kubernetes
already supports the label, but the feature gate is disabled in `kube-controller-manager`.
```

Reply:
Sorry, I'm still confused. Do you mean when the feature is re-enabled on kube-apiserver, but remains disabled on kube-controller-manager?

Reply:
I had in mind only the rollback part, so users are aware that it may cause clash of the new controllers. It might be better to just make a short mention in the `Can the feature be disabled once it has been enabled` section instead of here.

Reply:
I see, makes sense now, added a note along the lines to the suggested section. PTAL.

Reply:
looks good, but it may be better to be more concrete. 

`the service health` -> `the health of a job and its controller`

Reply:
much better, thanks